### PR TITLE
Add sorting logic by created_date for fetching triggers

### PR DESCRIPTION
**Pull request by utkarsharma2**
When fetching triggers we should ensure the old one gets picked up first, following FIFO. In this PR we have added a sorting logic by Trigger's class creation_date.

### CODE REVIEW

File: airflow/models/trigger.py [hunk 1]
```diff
@@ -220,16 +220,24 @@ def assign_unassigned(cls, triggerer_id, capacity, session: Session = NEW_SESSIO
...
+        if trigger_ids_query:
+            session.query(cls).filter(cls.id.in_([i.id for i in trigger_ids_query])).update(
+                {cls.triggerer_id: triggerer_id},
+                synchronize_session=False,
+            )
+        session.commit()
+
+    @classmethod
+    def get_sorted_triggers(cls, capacity, alive_triggerer_ids, session):
         trigger_ids_query = with_row_locks(
```
Comment (jedcunningham) [inline airflow/models/trigger.py]:
```suggestion
        return with_row_locks(
```

File: airflow/models/trigger.py [hunk 2]
```diff
@@ -220,16 +220,24 @@ def assign_unassigned(cls, triggerer_id, capacity, session: Session = NEW_SESSIO
...
             session,
             skip_locked=True,
         ).all()
-        if trigger_ids_query:
-            session.query(cls).filter(cls.id.in_([i.id for i in trigger_ids_query])).update(
-                {cls.triggerer_id: triggerer_id},
-                synchronize_session=False,
-            )
-        session.commit()
+        return trigger_ids_query
```
Comment (jedcunningham) [inline airflow/models/trigger.py]:
```suggestion
```

File: tests/models/test_trigger.py [hunk 3]
```diff
@@ -184,3 +185,34 @@ def test_assign_unassigned(session, create_task_instance):
...
+        ),
+    )
+    trigger_new.id = 2
+    session.add(trigger_old)
+    session.add(trigger_new)
+    session.commit()
+    assert session.query(Trigger).count() == 2
+
+    trigger_ids_query = Trigger.get_sorted_triggers(capacity=100, alive_triggerer_ids=[], session=session)
+    session.expire_all()
```
Comment (jedcunningham) [inline tests/models/test_trigger.py]: Do we need this?

Comment (utkarsharma2) [reply tests/models/test_trigger.py]: Nope, we don't need it. Removed it.

**Review by jedcunningham — APPROVED** [review]

**Pull request merged**